test/lsp_test_runner.cc

```diff
@@ -247,7 +251,14 @@ void testQuickFixCodeActions(LSPWrapper &lspWrapper, Expectations &test, const v
                                 applyCodeActionAssertions, error->toString(), false);
         }
 
-        for (auto codeActionAssertion : applyCodeActionAssertions) {
+        // This weird loop is here because `validateCodeActions` erases the elements from `applyCodeActionAssertions`
+        // and we are iterating over that container
+        auto applyCodeActionAssertionsSize = applyCodeActionAssertions.size();
+        for (int i = 0; i < applyCodeActionAssertionsSize; i++) {
```

Comment:
We don't actually use `i` anywhere. Let's drop the comment, and rewrite this to use

```cpp
while (!applyCodeActionAssertions.empty()) {
  auto initialSize = applyCodeActionAssertions.size();
  // ...
  ENFORCE(applyCodeActionAssertions.size() < initialSize);
}
```